### PR TITLE
cleanup: wrap the apiserver identity validation

### PR DESCRIPTION
**Pull request by pandaamanda**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kube-apiserver/app/options/validation.go [hunk 1]
```diff
@@ -150,6 +150,19 @@ func validateAPIPriorityAndFairness(options *ServerRunOptions) []error {
 	return nil
 }
 
+func validateAPIServerIdentity(options *ServerRunOptions) []error {
+	var errs []error
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerIdentity) {
```
Comment (caesarxuchao) [inline cmd/kube-apiserver/app/options/validation.go:155]: It's unnecessary to check the feature gate.

Comment (pandaamanda) [reply cmd/kube-apiserver/app/options/validation.go:155]: two flags valid when the feature gate enabled, so i add the logic.

Comment (pandaamanda) [reply cmd/kube-apiserver/app/options/validation.go]: fixed

**Pull request merged**